### DISCUSSION TITLE
Access to serializer/deserializer?

### DISCUSSION THREAD

**Discussion by PatrikN-work** [eclipse-langium/langium#1060] [Q&A] [answered]
Hi,

I'd like to use the langium as a serializer/deserializer for my semantic model , more specifically my GLSP semantic model.

Is there a (simple) way to access the langium serializer/deserializer "backend" programatically?

**Comment by msujew** [comment]
Hey @PatrikN-work,

you should be able to use the [`JsonSerializer`](https://github.com/langium/langium/blob/main/packages/langium/src/serializer/json-serializer.ts) service that performs both serialization and deserialization of AST models. Note that it's currently limited to references to elements that live in the same document.

**Comment by PatrikN-work** [comment]
Thanks for a quick anwer!

Sorry, I wasn't clear in my question. I want to serialize according to my Langium grammar not in json format (which I interpert that the JsonSeralizer does).

**Reply by spoenemann** [reply]
That feature is on our roadmap (#863), but we have no concrete plan when it will be ready. Until then you can implement a hard-coded serializer for your language in the same way as you would write a code generator.

**Reply by PatrikN-work** [reply]
Sorry, again I was unclear. I need seralization as well as deserailization.

Langium will have deserialization, I assume? 

Thanks!

**Reply by msujew** [reply]
With deserialization you just mean a parser, right? Langium of course provides a parser, given that it's a language engineering framework.

**Reply by PatrikN-work** [reply]
To be specific, Lexing and Parsing. Going from text to AST.

![image](https://github.com/langium/langium/assets/119344948/16096bc3-19d0-4d87-887c-88a5f5a39b45)

Is is possible to "hook into" that today and how?

**Reply by msujew** [reply, marked as answer]
Yes, you can just call the `LangiumParser` service of your language container:

https://github.com/langium/langium/blob/915ff5d53daf76415b0a838540c90e3437d3f8d0/packages/langium/src/parser/langium-parser.ts#L152-L164

**Comment by PatrikN-work** [comment]
Great, thanks. I'll give it a try. 

And please prioritize the serializer as well :-)
